### ISSUE TITLE
Specify a standard version of `clang-format`

### ISSUE THREAD

**Issue by wbernoudy** [dwavesystems/dwave-optimization#113]
Since we expect the C++ code to be formatted with `clang-format`, but there are inconsistencies between different versions, we should specify the version one is expected to use when doing development.

**Comment by arcondello** [comment]
We do not currently enforce clang-formatted code. We could consider doing so as well.

**Comment by wbernoudy** [comment]
Agreed! For now at least, it would be nice to just have a note in the readme.

**Comment by arcondello** [comment]
I don't think this is easy.

The ecosystem as a whole doesn't have a lot of consensus.

* VSCode uses 18.1.7 (I think?) https://github.com/microsoft/vscode-cpptools/blob/1e89750b0bb9629ba0262a6eb44d1ab41eb77d51/Extension/CHANGELOG.md?plain=1#L137
* `apt install clang format` pulls in 14.0.0
* The LLVM source tree itself doesn't seem to be well formatted, albeit as of [March 2022](https://clang.llvm.org/docs/ClangFormattedStatus.html).

For internal repos, I am reasonably comfortable enforcing a version. But for public repos like this, where we want to encourage contribution from users, I don't think it's reasonable to force folks to install a specific version of clang simply to make a PR against this repo.

**Comment by arcondello** [comment]
FWIW, https://github.com/ssciwr/clang-format-wheel feels like it might actually help. Could make it part of the `dev` dependencies and enforce a version.

**Comment by arcondello** [comment]
Also has a pre-commit option: https://github.com/ssciwr/clang-format-wheel#use-from-pre-commit
